frontend/test/metabase/scenarios/dashboard/action-permissions.cy.spec.js

```diff
@@ -0,0 +1,56 @@
+import { onlyOn } from "@cypress/skip-test";
+import { restore, popover } from "__support__/cypress";
+
+// TODO Add other permissions
+const PERMISSIONS = {
+  curate: ["admin", "normal"],
+  view: ["readonly"],
+};
+
+describe("dashboard action permissions", () => {
+  beforeEach(() => {
+    restore();
+  });
+
+  Object.entries(PERMISSIONS).forEach(([permission, userGroup]) => {
+    context(`${permission} access`, () => {
+      userGroup.forEach(user => {
+        beforeEach(() => {
+          cy.signIn(user);
+          cy.visit("/dashboard/1");
+        });
```

Comment:
@nemanjaglumac I borrowed that approach from [collections test suite](https://github.com/metabase/metabase/blob/4b393e9ea2c5bbb66048048ace29b7b378ed4686/frontend/test/metabase/scenarios/collections/permissions.cy.spec.js). It's very helpful and it looks like we'd need to run many tests for users with different permissions. Maybe it's worth to extract as a shared utility?

Reply:
I am glad to hear it's helpful. Permissions testing is still a WIP, so I'd maybe wait a bit longer before we decide which parts exactly to extract and reuse.
It's a bit more manual wok until then but we might end up optimizing too early.

Btw, unless I'm missing something, the "duplicate" dashboard part for `curate` group was already covered in `frontend/test/metabase/scenarios/collections/permissions.cy.spec.js`.

I'm sorry that file is a mess currently (like I said it's still WIP) - we'll probably split and divide it in the end. But for now, I'd probably just add tests there.

Reply:
Agree, let's keep an eye on it

Reply:
> Btw, unless I'm missing something, the "duplicate" dashboard part for curate group was already covered

Yes, it's there. I just added a test that ensures "duplicate" action isn't displayed to user without a _write_ permission

Reply:
> I just added a test that ensures "duplicate" action isn't displayed to user without a write permission

Exactly. 
that's why I think it's an overkill to add a separate file for it when we already have `frontend/test/metabase/scenarios/collections/permissions.cy.spec.js`.

Reply:
Yeah, the "Duplicate dashboard" test for the collection page is in `frontend/test/metabase/scenarios/collections/permissions.cy.spec.js`

The code in this thread tests "Duplicate" is hidden from the Dashboard page (`/dashboard/:id`). The test is in `frontend/test/metabase/scenarios/dashboard`

Reply:
There is the whole section dedicated to dashboard edit menu if you visit the dashboard page `/dashboard/:id`.
https://github.com/metabase/metabase/blob/5117a252c6b74b4f73ccecab6fb138914517197b/frontend/test/metabase/scenarios/collections/permissions.cy.spec.js#L290

We will divide that file later, but for now while it is still a work in progress let's keep everything in one place, please.

Reply:
Sure, thank you